driver-core/src/main/com/mongodb/operation/QueryBatchCursor.java

```diff
@@ -58,7 +58,7 @@
         this.namespace = firstQueryResult.getNamespace();
         this.limit = limit;
         this.batchSize = batchSize;
-        this.decoder = notNull("decoder", decoder);
+        this.decoder = decoder;
```

Comment:
Because there is no get-more for distinct.  Got it.  It's a bit of a wart though.